src/ripple/rpc/handlers/GetAggregatePrice.cpp

```diff
@@ -85,18 +85,18 @@ iteratePriceData(
 
         auto const ledger = context.ledgerMaster.getLedgerBySeq(prevSeq);
         if (!ledger)
-            return;
+            return;  // LCOV_EXCL_LINE
 
         meta = ledger->txRead(prevTx).second;
 
+        prevChain = chain;
```

Comment:
What does moving this line do? I don't really understand what `prevChain` is doing, and there don't appear to be any tests checking this behavior.

Reply:
This handles an unlikely case when `CreatedNode/ModifiedNode` not found in the inner loop for `AffectedNodes`. If this happens then `prevChain` is going to be equal to `chain` and the iteration stops. I don't know if it's possible to reproduce it unless the ledger state is changed in a way, which is not possible to do with the transactions.

Reply:
Does moving this line of code make any difference other than the slight efficiency improvement from running fewer times?

Reply:
`prevChain` has to be assigned outside of the inner loop. Initially `prevChain` is `nullptr` and `chain` points to `oracle`.  If there is no `AffectedNodes` then both `prevChain` and `chain` are unchanged and the check `(prevChain == chain)` is not going to fail.